File: easy/destination_city/destination_city.go
package main

import "fmt"

func destCity(paths [][]string) string {
	citiesWithOutgoingPaths := make(map[string]bool)
	for _, path := range paths {
		citiesWithOutgoingPaths[path[0]] = true
		if _, ok := citiesWithOutgoingPaths[path[1]]; !ok {
			citiesWithOutgoingPaths[path[1]] = false
		}
	}

	for city, isSource := range citiesWithOutgoingPaths {
		if !isSource {
			return city
		}
	}
	return ""
}

func main() {
	fmt.Println("Solution 1 : ", destCity([][]string{{"London", "New York"}, {"New York", "Lima"}, {"Lima", "Sao Paulo"}}))
	fmt.Println("Solution 2 : ", destCity([][]string{{"B", "C"}, {"D", "B"}, {"C", "A"}}))
	fmt.Println("Solution 3 : ", destCity([][]string{{"A", "Z"}}))
}

File: easy/destination_city/destination_city.py
class Solution(object):
    def destCity(self, paths):
        """
        :type paths: List[List[str]]
        :rtype: str
        """
        cities_with_outgoing_paths = {}
        for path in paths:
            cities_with_outgoing_paths[path[0]] = True
            if not path[1] in cities_with_outgoing_paths:
                cities_with_outgoing_paths[path[1]] = False

        for city, is_source in cities_with_outgoing_paths.items():
            if not is_source:
                return city


s = Solution()
print("Solution 1 : ", s.destCity(
    [["London", "New York"], ["New York", "Lima"], ["Lima", "Sao Paulo"]]))
print("Solution 2 : ", s.destCity([["B", "C"], ["D", "B"], ["C", "A"]]))
print("Solution 3 : ", s.destCity([["A", "Z"]]))
